scripts/release/release_notes.py

```diff
@@ -0,0 +1,132 @@
+) -> str:
+    f"""Generate a release notes based on the changes since the previous version tag.
+
+    Parameters:
+        repository_path: Path to the Git repository. Default is the current directory '.'.
+        changelog_sub_path: Path to the changelog directory relative to the repository root. Default is '{DEFAULT_CHANGELOG_PATH}.
+        initial_commit_sha: SHA of the initial commit to start from if no previous version tag is found.
+        initial_version: Version to use if no previous version tag is found. Default is '{DEFAULT_INITIAL_GIT_TAG_VERSION}'.
+
+    Returns:
+        Formatted release notes as a string.
+    """
+    repo = Repo(repository_path)
+
+    version, changelog = calculate_next_version_with_changelog(
+        repo, changelog_sub_path, initial_commit_sha, initial_version
+    )
+
+    with open("scripts/release/release_notes_tpl.md", "r") as f:
+        template = Template(f.read())
+
+    parameters = {
+        "version": version,
+        "preludes": [c.contents for c in changelog if c.kind == ChangeKind.PRELUDE],
+        "breaking_changes": [c.contents for c in changelog if c.kind == ChangeKind.BREAKING],
+        "features": [c.contents for c in changelog if c.kind == ChangeKind.FEATURE],
+        "fixes": [c.contents for c in changelog if c.kind == ChangeKind.FIX],
+        "others": [c.contents for c in changelog if c.kind == ChangeKind.OTHER],
+    }
+
+    return template.render(parameters)
+
+
+def calculate_next_version_with_changelog(
+    repo: Repo, changelog_sub_path: str, initial_commit_sha: str | None, initial_version: str
+) -> (str, list[ChangeEntry]):
+    previous_version_tag, previous_version_commit = find_previous_version(repo, initial_commit_sha)
+
+    changelog: list[ChangeEntry] = get_changelog_entries(previous_version_commit, repo, changelog_sub_path)
+    changelog_kinds = list(set(entry.kind for entry in changelog))
+
+    # If there is no previous version tag, we start with the initial version tag
+    if not previous_version_tag:
+        version = initial_version
+    else:
+        version = calculate_next_release_version(previous_version_tag.name, changelog_kinds)
+
+    return version, changelog
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Generate release notes based on the changes since the previous version tag.",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "-p",
+        "--path",
+        default=".",
+        metavar="",
+        action="store",
+        type=pathlib.Path,
+        help="Path to the Git repository. Default is the current directory '.'",
+    )
+    parser.add_argument(
+        "-c",
+        "--changelog-path",
+        default=DEFAULT_CHANGELOG_PATH,
+        metavar="",
+        action="store",
+        type=str,
+        help=f"Path to the changelog directory relative to the repository root. Default is '{DEFAULT_CHANGELOG_PATH}'",
```

Comment:
Nit: this is not factually correct. Repository root is not involved here as far as see.

```suggestion
        help=f"Path to the changelog directory relative to a current working directory. Default is '{DEFAULT_CHANGELOG_PATH}'",
```